### PR TITLE
Gitea helm

### PR DESCRIPTION
**Pull request by mbaldessari**

### CODE REVIEW

File: controllers/utils.go [hunk 1]
```diff
@@ -188,3 +198,27 @@ func compareMaps(m1, m2 map[string][]byte) bool {
 
 	return true
 }
+
```
Comment (mbaldessari) [inline controllers/utils.go:201]: I would be very weary of rolling our own password generating code (even if it is a first time pass for admin). Maybe something simpler like:
```
func GenerateRandomPassword(length int) (string, error) {
    bytes := make([]byte, length)
    _, err := rand.Read(bytes)
    if err != nil {
        return "", err
    }
    return base64.URLEncoding.EncodeToString(bytes), nil
}
```

Also in this case we really want to use `crypto/rand` and not `math/rand`

File: go.mod [hunk 2]
```diff
@@ -11,21 +11,26 @@ require (
...
 	github.com/operator-framework/operator-lifecycle-manager v0.20.0
 	github.com/segmentio/analytics-go/v3 v3.3.0
 	go.uber.org/mock v0.4.0
-	golang.org/x/crypto v0.17.0
 	gopkg.in/yaml.v3 v3.0.1
 	helm.sh/helm/v3 v3.9.4
-	k8s.io/api v0.28.4
-	k8s.io/apimachinery v0.28.4
+	k8s.io/api v0.29.0
+	k8s.io/apimachinery v0.29.0
```
Comment (mbaldessari) [inline go.mod:23]: Not needed as it will be replaced below anyway

File: go.mod [hunk 3]
```diff
@@ -11,21 +11,26 @@ require (
 	github.com/google/uuid v1.5.0
 	github.com/onsi/ginkgo/v2 v2.13.2
 	github.com/onsi/gomega v1.30.0
-	github.com/openshift/api v0.0.0-20231130225947-f2209623068b
+	github.com/openshift/api v0.0.0-20240118221011-59fa376de7cb
```
Comment (mbaldessari) [inline go.mod:14]: Is this update needed? If not best  keep it for a separate pr?

File: api/v1alpha1/giteaserver_types.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+type GiteaServerSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Helm Chart URL. Default value: https://charts.validatedpatterns.io
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=1
+	HelmChartUrl string `json:"helmChartUrl,omitempty"`
+	// Namespace where helm chart will be deployed to. Default: gitea
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,order=2
+	Namespace string `json:"namespace,omitempty"`
```
[the remark this answers is not in the document]
Comment (claudiol) [reply api/v1alpha1/giteaserver_types.go:36]: Yeah that is a great idea.  I was conflicting in having the namespace available to change.  I will drop the namespace out of the CRD and use the vp-gitea namespace.
